code/game/objects/items/robot/robot_parts.dm

```diff
@@ -62,6 +62,59 @@
 				return 1
 	return 0
 
+/obj/item/robot_suit/wrench_act(mob/living/user, obj/item/I) //Deconstucts empty borg shell. Flashes remain unbroken because they haven't been used yet
+	var/turf/T = get_turf(src)
+	forceMove(T)
+	if(l_leg || r_leg || chest || l_arm || r_arm || head)
+		if(I.use_tool(src, user, 5, volume=50))
+			if(l_leg)
+				l_leg.forceMove(T)
+				l_leg = null
+			if(r_leg)
+				r_leg.forceMove(T)
+				r_leg = null
+			if(chest)
+				if (chest.cell) //Sanity check.
+					chest.cell.forceMove(T)
+					chest.cell = null
+				chest.forceMove(T)
+				new /obj/item/stack/cable_coil(T, 1)
+				chest.wired = 0
+				chest = null
+			if(l_arm)
+				l_arm.forceMove(T)
+				l_arm = null
+			if(r_arm)
+				r_arm.forceMove(T)
+				r_arm = null
+			if(head)
+				head.forceMove(T)
+				head.flash1.forceMove(T)
+				head.flash1 = null
+				head.flash2.forceMove(T)
+				head.flash2 = null
+				head = null
+			to_chat(user, "<span class='notice'>You disassemble the cyborg shell.</span>")
+	else
+		to_chat(user, "<span class='notice'>There is nothing to remove from the endoskeleton.</span>")
+	updateicon()
+
+/obj/item/robot_suit/screwdriver_act(mob/living/user, obj/item/I) //Swaps the power cell if you're holding a new one in your other hand.
+	var/obj/item/stock_parts/cell/temp_cell = user.is_holding_item_of_type(/obj/item/stock_parts/cell)
+	if(QDELETED(temp_cell))
+		return
+	if(QDELETED(chest.cell))
+		to_chat(user, "<span class='notice'>There doesn't seem to unscrew from [src].</span>")
+		return
+	if(!user.transferItemToLoc(temp_cell, chest))
+		to_chat(user, "<span class='warning'>[temp_cell] is stuck to your hand, you can't put it in [src]!</span>")
+		return
+	if(!user.put_in_hands(chest.cell))
+		chest.cell.forceMove(drop_location())
+	chest.cell = temp_cell
+	to_chat(user, "<span class='notice'>You replace [src]'s power cell.</span>")
+	return TRUE

```

Comment:
Actually, rethought this a bit. Why not just make it so you can always remove the powercell from the shell?

```
/obj/item/robot_suit/proc/put_in_hand_or_drop(mob/living/user, obj/item/I) //normal put_in_hands() drops the item ontop of the player, this drops it at the suit's loc
	if(!user.put_in_hands(I))
		I.forceMove(drop_location())
		return FALSE
	return TRUE

/obj/item/robot_suit/screwdriver_act(mob/living/user, obj/item/I) //Removes the current power cell, or swaps it if you're holding a new one.
	if(QDELETED(chest) || QDELETED(chest.cell))
		to_chat(user, "<span class='notice'>There doesn't seem to be anything to unscrew from [src].</span>")
		return

	var/obj/item/stock_parts/cell/temp_cell = user.is_holding_item_of_type(/obj/item/stock_parts/cell)
	var/swap_failed
	if(QDELETED(temp_cell)) //if we're not holding a cell
		swap_failed = TRUE
	else if(!user.transferItemToLoc(temp_cell, chest))
		swap_failed = TRUE
		to_chat(user, "<span class='warning'>[temp_cell] is stuck to your hand, you can't put it in [src]!</span>")

	put_in_hand_or_drop(user, chest.cell)

	if(swap_failed)
		to_chat(user, "<span class='notice'>You remove [chest.cell] from [src].</span>")
		chest.cell = null
		return

	chest.cell = temp_cell
	to_chat(user, "<span class='notice'>You replace [src]'s power cell.</span>")
	return TRUE
```

Reply:
Presumably, the reason you can't make a cell-less shell in the first place is because it's nice as the borg player to _know_ that you won't spawn in with power problems. Sure, the robo player might use a depleted cell, but the borg can fix that themselves with a charging station, rather than having to sit there and ping or buzz until someone bothers to figure out what they need.

If you specifically need to use the cell somewhere quickly, the disassemble-via-wrench will be available, same as if you suddenly need to give someone a prosthetic limb.

Reply:
Just let the ghost know how much power there is then if you're worried about that

Reply:
Ghosts don't go in borgs, they go in MMIs that go into borgs. The ghost doesn't have an option in this situation outside of not opting into borgs at all.

I can assure you that issue won't happen anymore than roboticists immediately flashing the borg on spawn to remove their powercell.

Reply:
Seems I'm outvoted. I'll add the code in a bit, after I test and reorganize it a bit.

Would anyone be opposed to me preventing an MMI to be inserted without a cell in the shell?